cloudprober/templates/deployment.yaml

```diff
@@ -48,10 +48,19 @@ spec:
             {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-          args: ['--config_file', '/cfg/cloudprober.{{- .Values.configFormat -}}', '--logtostderr']
+          args:
+            - "--logtostderr"          
+            {{ if .Values.cloudprober_args }}
+            {{ with .Values.cloudprober_args }}
+              - {{- toYaml . | nindent 12 }}
+            {{ else }}
+              - "--config_file"
+              - "{{ .Values.cloudprober_cfg_path | default "/cfg"  }}/cloudprober.{{- .Values.configFormat -}}"
+            {{- end }}
+            {{- end }}
```

Comment:
So I like the idea of extraArgs in #35, i.e. you always have the config_file arg. If we have disable that ever, there should like be another template variable for that.

and, for extraArgs, something like this will likely work better (i.e. explicitly iterate over them instead of indenting the input array.
https://github.com/grafana/helm-charts/blob/4401adc2780fba7a5a8da3dcda1a2d6fb039bccc/charts/grafana/templates/_pod.tpl#L945

Reply:
I'm ok with extraArgs, but I would like a way to override the name of the config file. A variable to set a config file name suffix would also work for me

My use case is that I have multiple cloudprober_${environment} files, since I'm running Cloudprober both in and outside of K8s

Reply:
* I was wondering why you'd want to override the config path. This seems entirely internal to the helm chart.
* If you're passing the configMap (i.e. helm is not creating the configMap), config can be named differently. So, adding a variable called 'configFileName' will likely make sense. We can use it in deployment.yaml and here: https://github.com/cloudprober/helm-charts/blob/9363f8ffba44f8c12df11b268e85a029e64e23a1/cloudprober/templates/configmap.yaml#L7
(I am guessing that's the use case you described in the comment?).

Reply:
> * I was wondering why you'd want to override the config path. This seems entirely internal to the helm chart.

The config path issue is due to inconsistencies between the config include statement and the file targets. Config includes can take relative paths, but the file based targets require an absolute path. I can workaround this by symlinking /cfg  to /etc/cloudprober in my non k8s environments.

> * If you're passing the configMap (i.e. helm is not creating the configMap), config can be named differently. So, adding a variable called 'configFileName' will likely make sense. We can use it in deployment.yaml and here: https://github.com/cloudprober/helm-charts/blob/9363f8ffba44f8c12df11b268e85a029e64e23a1/cloudprober/templates/configmap.yaml#L7

Correct, this is my problem. I generate my configMap and pass it in, but can never use it because I don't have a cloudprober.cfg file within it.

I've created the configFileName variable as you've suggested and added it to both configmap.yaml and deployment.yaml